java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -43,32 +43,50 @@ public class ColumnView implements AutoCloseable, BinaryOperable {
   protected final ColumnVector.OffHeapState offHeap;
 
   /**
-   * Constructs a Column View given a native view address
+   * Constructs a Column View given a native view address. This asserts that if the ColumnView is
+   * of nested-type it doesn't contain non-empty nulls
    * @param address the view handle
+   * @throws AssertionError if the address points to a nested-type view with non-empty nulls
    */
   ColumnView(long address) {
     this.viewHandle = address;
     this.type = DType.fromNative(ColumnView.getNativeTypeId(viewHandle), ColumnView.getNativeTypeScale(viewHandle));
     this.rows = ColumnView.getNativeRowCount(viewHandle);
     this.nullCount = ColumnView.getNativeNullCount(viewHandle);
     this.offHeap = null;
-    AssertEmptyNulls.assertNullsAreEmpty(this);
+    try {
```

Comment:
This covers the single `ColumnView` case. Note that there is one more case we should cover (that I can see in cuDF): `ColumnView.splitAsViews`. This function iterates over handles and instantiates the `ColumnView`s in a loop.

If one of those iterations asserts, then we leak the handles.

I think what can be done in that case is to declare a `Throwable` object (but not instantiated) before going into the loop, and set it to the first assertion in a try/catch. If you see an assertion while instantiating CV `i`, you know you have dealt with this particular handle, so what you need to do is then close out the remaining handles at: `[i+1, nativeHandles.length)`,  and `[0, i-1]`.

Reply:
Why just handle this specific case? Why not just catch the `Throwable` and just close everything out and exit the method after rethrowing?

Reply:
I am fine if you want to change this to catch Throwable, I think you are looking at the other method calls in this constructor as they could throw as well? As long as you think they can throw, it would be a good change IMHO.